src/core/response/parameters.rs

```diff
@@ -46,28 +52,96 @@ impl From<VpToken> for Json {
     }
 }
 
+impl VpToken {
+    /// Parse the VP Token as a JSON object.
+    ///
+    /// This will attempt to decode the token as base64, and if that fails, it
+    /// will attempt to parse the token as a JSON object.
+    ///
+    /// See: https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#section-6.1-2.2
+    ///
+    /// If you want to check for decode errors, use [VpToken::decode_base64].
+    pub fn parse(&self) -> Result<Json, Error> {
+        match self.decode_base64() {
+            Ok(decoded) => Ok(decoded),
+            Err(_) => Ok(serde_json::from_str(&self.0)?),
+        }
+    }
+
+    /// Return the Verifiable Presentation Token as a JSON object from a base64
+    /// encoded string.
+    pub fn decode_base64(&self) -> Result<Json, Error> {
+        let decoded = BASE64_STANDARD.decode(&self.0)?;
+        Ok(serde_json::from_slice(&decoded)?)
+    }
```

Comment:
These methods are inaccurate:

> JSON String or JSON object that MUST contain a single Verifiable Presentation or an array of JSON Strings and JSON objects each of them containing a Verifiable Presentations. Each Verifiable Presentation MUST be represented as a JSON string (that is a Base64url encoded value) or a JSON object depending on a format as defined in Annex E of [[OpenID.VCI](https://openid.net/specs/openid-4-verifiable-presentations-1_0-20.html#OpenID.VCI)]

We have a bug in our definition of `VpToken`. It should be something like:

```rust
enum VpToken {
    Single(String),
    SingleAsMap(Map<String, Value>),
    Many(Vec<VpToken>),
}
```

Reply:
I have a TODO in the code to update the VpToken. Maybe we can do this in another PR.